File: codeforces/problemset/455A.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
#define INF 100005
#define ll long long
int main() {
    int n;
    cin >> n;
    vector<ll> a, cnt(INF+1,0);
    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;
        a.push_back(x);
        cnt[x]++;
    }
    // dp[x] is the max value if x was the last number picked
    vector<ll> dp(INF+1,0);
    dp[a[0]] = a[0] * cnt[a[0]];
    for (int i = 1; i < 100000 + 1; i++) {
        dp[i] = max(cnt[i]*i + dp[max(0,i-2)], dp[i-1]);
    }
    cout << *max_element(dp.begin(), dp.end()) << endl;

    return 0;
}

File: codeforces/problemset/455A.py
n = int(input())
cnt = [0] * (10**5+5)
a = list(map(int, input().split()))
for x in a:
    cnt[x] += 1
# dp[i] = max val if i was the last number picked
dp = [0] * (10**5+5) 
dp[a[0]] = cnt[a[0]]*a[0]
for i in range(1, n+1):
        dp[i] = max(cnt[i]*i + dp[i-2], dp[i-1])
print(max(dp))
